HIP-Basic/moving_average/moving_average_vs2017.vcxproj

```diff
@@ -65,6 +65,12 @@
   <PropertyGroup Label="HIP_clang" Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
     <OffloadArch>gfx1030;gfx1100;gfx1101;gfx1102</OffloadArch>
   </PropertyGroup>
+  <PropertyGroup Label="HIP" Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
+    <NvccAdditionalOptions>-std=c++17</NvccAdditionalOptions>
+  </PropertyGroup>
+  <PropertyGroup Label="HIP" Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
+    <NvccAdditionalOptions>-std=c++17</NvccAdditionalOptions>
+  </PropertyGroup>
```

Comment:
C++ `LanguageStandard` is implemented for `nvcc` finally, so better to set it in the corresponding PPP, for instance:

```xml
  <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64' and '$(PlatformToolset)'=='HIP_nvcc'">
    <ClCompile>
      <PreprocessorDefinitions>__CUDACC__;_DEBUG;_CONSOLE;%(PreprocessorDefinitions)</PreprocessorDefinitions>
      <LanguageStandard>stdcpp17</LanguageStandard>
    </ClCompile>
  </ItemDefinitionGroup> 
```
Btw, what about `HIP_clang`, is it ok with the default `c++14`?

Reply:
Of course.
Compilation errors fixed by this PR are all related to host code. Unlike Clang, in the case of compiling host code, nvcc uses the system's native compiler: Microsoft cl in this case. 
Using the c++17 standard is required by cl but not by Clang.

Reply:
Thanks, but 1) better to have an error in the PR description as well 2) it's a bit strange that for `clang` it is enough to have c++14, and for `cl` is not; sounds like some `c++17` feature is introduced in `c++14` for clang - if it is so, it is a very unusual situation.

Reply:
@saadrahim, can HIP sources be touched in order to fix the `nvcc` path only?